components/logins/src/migrate_sqlcipher_db.rs

```diff
@@ -67,27 +101,80 @@ fn sqlcipher_3_compat(conn: &Connection) -> Result<()> {
     Ok(())
 }
 
-fn sqlcipher_export(conn: &mut Connection, new_db_path: impl AsRef<Path>) -> Result<()> {
-    // Export sqlite data to a plaintext DB using the strategy from example 2 here:
-    // https://www.zetetic.net/sqlcipher/sqlcipher-api/index.html#sqlcipher_export
-    let path = new_db_path.as_ref().as_os_str();
-    let path = path
-        .to_str()
-        .ok_or_else(|| ErrorKind::InvalidPath(path.to_os_string()))?;
-
-    conn.execute("ATTACH DATABASE ? AS plaintext key ''", &[path])?;
-    // this one is a bit weird because it's a SELECT statement that we know will return 0 rows.
-    // rusqlite will return an Error if we use execute(), so we use query_row with a dummy closure
-    conn.query_row("SELECT sqlcipher_export('plaintext')", NO_PARAMS, |_| Ok(0))?;
-    conn.execute("DETACH DATABASE plaintext", NO_PARAMS)?;
-    Ok(())
+//Manually copy over row by row from sqlcipher db to a plaintext db
+pub fn migrate_from_sqlcipher_db(
```

Comment:
Is this (or will this be) the `migrate_single_row` you were referring to in the PR description?

Reply:
ah I refactored a bit so `migrate_single_row` doesn't exist anymore. However this will still be the main entrypoint